### ISSUE TITLE
Add "save changes" warning in Grapher admin

### ISSUE THREAD

**Issue by paarriagadap** [owid/owid-grapher#3594]
## Core problem

Many times while working in Grapher admin we are not entirely sure if we have saved the latest changes, as there is no indication of it. We could close the window and lose these changes inadvertently. At least in my case, I press Update chart _just in case_ I forget to do it, which saves a new version indistinguishable from the previous one: as a consequence, the _Revisions_ list grows needlessly.

## Proposed solution

Add a warning in case the user has not saved relevant changes and they are closing the window, something along the lines of any typical software, like:
<img width="246" alt="image" src="https://github.com/owid/owid-grapher/assets/63430031/4d23d2f6-0b4f-432c-a4a4-c2966b7e49ce">

## Alternatives

- Using a visual reference in the window for changes not saved, like 
<img width="271" alt="image" src="https://github.com/owid/owid-grapher/assets/63430031/13ea9c62-104e-475e-8b84-a96c4c1c4d2f">

- Saving temporary changes automatically could also be an option, but not saving to live, as changes can be unfinished.

**Comment by danyx23** [comment]
Solving this can be a bit tricky because if you get even one property in the admin wrong it can mean that either you can't save actual changes or you get spurious change indicator warnings.

Another alternative would be to modify the function that saves charts and just do nothing if the old and new chart config are exactly the same. This would mean you would still press save often but it would not create new revisions unless they are needed.

**Comment by paarriagadap** [comment]
@danyx23 Yeah, that can work to avoid unnecessary versions. Thanks for taking a look!

**Comment by paarriagadap** [comment]
@danyx23 What you say can also help to avoid sending unnecessary requests to the deploy queue, so it can be useful for that side as well.

**Comment by sophiamersmann** [comment]
I remember fixing this a while ago when I refactored the chart editor. I just tested it and it seems to work, i.e. this pop-up 👇🏻 shows up iff navigating away with unsaved changes (it doesn't show when there aren't unsaved changes).

@paarriagadap please re-open if the issue persist.

<img width="1510" alt="Image" src="https://github.com/user-attachments/assets/3d902eb1-2304-49c8-a7df-19750f3a50a8" />

**Comment by paarriagadap** [comment]
@sophiamersmann Ah, that works, thanks! I assume that doing this when closing the tab is much trickier, or not possible, right?

**Comment by sophiamersmann** [comment]
@paarriagadap it shouldn't be too hard. I tried it [here](http://staging-site-admin-unsaved-changes/admin). Does that work for you? It should warn you when you close the tab or refresh.

**Comment by paarriagadap** [comment]
@sophiamersmann Thanks for taking a look! It works fine when I toggle the map view, but it doesn't when, for example, I change the display name of an indicator. Do you know the reason why?

No rush with this of course, have a good time with the team in Alella!
